components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.endpoint/src/main/java/org/wso2/carbon/identity/configuration/mgt/endpoint/impl/ResourceApiServiceImpl.java

```diff
@@ -0,0 +1,242 @@
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.configuration.mgt.core.exception.ConfigurationManagementClientException;
+import org.wso2.carbon.identity.configuration.mgt.core.exception.ConfigurationManagementException;
+import org.wso2.carbon.identity.configuration.mgt.core.model.Attribute;
+import org.wso2.carbon.identity.configuration.mgt.core.model.Resource;
+import org.wso2.carbon.identity.configuration.mgt.core.model.Resources;
+import org.wso2.carbon.identity.configuration.mgt.endpoint.ResourceApiService;
+import org.wso2.carbon.identity.configuration.mgt.endpoint.dto.AttributeDTO;
+import org.wso2.carbon.identity.configuration.mgt.endpoint.dto.ResourceAddDTO;
+import org.wso2.carbon.identity.configuration.mgt.endpoint.dto.ResourcesDTO;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import javax.ws.rs.core.Response;
+
+import static org.wso2.carbon.identity.configuration.mgt.core.constant.ConfigurationConstants.RESOURCE_PATH;
+import static org.wso2.carbon.identity.configuration.mgt.endpoint.util.ConfigurationEndpointUtils.getAttributeDTO;
+import static org.wso2.carbon.identity.configuration.mgt.endpoint.util.ConfigurationEndpointUtils.getAttributeFromDTO;
+import static org.wso2.carbon.identity.configuration.mgt.endpoint.util.ConfigurationEndpointUtils.getConfigurationManager;
+import static org.wso2.carbon.identity.configuration.mgt.endpoint.util.ConfigurationEndpointUtils.getOperationNotSupportedDTO;
+import static org.wso2.carbon.identity.configuration.mgt.endpoint.util.ConfigurationEndpointUtils.getResourceAddFromDTO;
+import static org.wso2.carbon.identity.configuration.mgt.endpoint.util.ConfigurationEndpointUtils.getResourceDTO;
+import static org.wso2.carbon.identity.configuration.mgt.endpoint.util.ConfigurationEndpointUtils.getResourcesDTO;
+import static org.wso2.carbon.identity.configuration.mgt.endpoint.util.ConfigurationEndpointUtils.handleBadRequestResponse;
+import static org.wso2.carbon.identity.configuration.mgt.endpoint.util.ConfigurationEndpointUtils.handleServerErrorResponse;
+import static org.wso2.carbon.identity.configuration.mgt.endpoint.util.ConfigurationEndpointUtils.handleUnexpectedServerError;
+
+public class ResourceApiServiceImpl extends ResourceApiService {
+
+    private static final Log LOG = LogFactory.getLog(ResourceApiServiceImpl.class);
+
+    @Override
+    public Response resourceGet() {
+
+        try {
+            Resources resources = getConfigurationManager().getResources();
+            ResourcesDTO resourcesDTO = getResourcesDTO(resources);
+            return Response.ok().entity(resourcesDTO).build();
+        } catch (ConfigurationManagementClientException e) {
+            return handleBadRequestResponse(e, LOG);
+        } catch (ConfigurationManagementException e) {
+            return handleServerErrorResponse(e, LOG);
+        } catch (Throwable throwable) {
```

Comment:
Shall we use Exception here? Throwable will catch runtime exceptions also.

Reply:
This is the expected behavior since both checked and unchecked exceptions are caught and handled as unexpected server errors.